### DISCUSSION TITLE
Fulltext search on MongoDB?

### DISCUSSION THREAD

**Discussion by iyobo** [marcj/deepkit#532] [General]
I am interested in seeing an example of how to perform full text search on Mongodb. The documentation on the mongoDB side of things seems sparse.

Full text search has 2 phases. Indexing, and searching.

## Indexing
Though I am not 100% sure, I assume indexing for fulltextsearch can be achieved with the following to search across 2 fields called name and meta. Please confirm.
```
@entity.name('user').index(['name', 'meta'], { fulltext: true })
export class User 
```
## Searching
I'm lost on this one. It seems like it would be a raw query, but I do not see any examples of how to do anything raw with mongodb.

**Comment by iyobo** [comment]
It would seem not even the indexing portion of this is working. I don't see any text based indexes being created.

Please let me know if these features are supported or doable with deepkit?

**Reply by iyobo** [reply]
@marcus-sa please comment.

**Reply by marcj** [reply]
You have to manually call `await database.migrate()` for MongoDB. This way the indexes defined in your types/entities are synced with the actual configured mongodb database.
